pkgs/development/compilers/rust/rustc.nix

```diff
@@ -52,6 +42,9 @@ stdenv.mkDerivation {
   # versions.
   RUSTC_BOOTSTRAP = "1";
 
+  # Increase codegen units to introduce parallelism within the compiler.
+  RUSTFLAGS = "-Ccodegen-units=10";
```

Comment:
That won't help, you'll still have to disable it before the `checkPhase`. And this is more related to `--cores` then parallel building, it's still a single `rustc` process if you use this with parallel building enabled.